diff --git a/lib/index.d.ts b/lib/index.d.ts
index 89ebc1cb09648b5c9665306b2c1e3013aebf0706..0c5513dc4934e6b05ddda2a9958604aa08c2b85d 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -6,6 +6,7 @@ declare type Padding<T> = T | {
     left?: T;
 };
 declare type Props = React.HTMLAttributes<HTMLDivElement> & {
+    language: string;
     value: string;
     onValueChange: (value: string) => void;
     highlight: (value: string) => string | React.ReactNode;
diff --git a/lib/index.js b/lib/index.js
index e01824d041688130d1c733882fa4e4f9a458e520..7acfcaab66d2ca1a8031d0341f42d220cc0155b3 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -429,7 +429,7 @@ var Editor = /** @class */ (function (_super) {
         };
         var highlighted = highlight(value);
         return (React.createElement("div", __assign({}, rest, { style: __assign(__assign({}, styles.container), style) }),
-            React.createElement("pre", __assign({ className: preClassName, "aria-hidden": "true", style: __assign(__assign(__assign({}, styles.editor), styles.highlight), contentStyle) }, (typeof highlighted === 'string'
+            React.createElement("pre", __assign({ className: preClassName, "aria-hidden": "true", "data-language": _a.language, "data-evaluate-disabled": "true", style: __assign(__assign(__assign({}, styles.editor), styles.highlight), contentStyle) }, (typeof highlighted === 'string'
                 ? { dangerouslySetInnerHTML: { __html: highlighted + '<br />' } }
                 : { children: highlighted }))),
             React.createElement("textarea", { ref: function (c) { return (_this._input = c); }, style: __assign(__assign(__assign({}, styles.editor), styles.textarea), contentStyle), className: className + (textareaClassName ? " ".concat(textareaClassName) : ''), id: textareaId, value: value, onChange: this._handleChange, onKeyDown: this._handleKeyDown, onClick: onClick, onKeyUp: onKeyUp, onFocus: onFocus, onBlur: onBlur, disabled: disabled, form: form, maxLength: maxLength, minLength: minLength, name: name, placeholder: placeholder, readOnly: readOnly, required: required, autoFocus: autoFocus, autoCapitalize: "off", autoComplete: "off", autoCorrect: "off", spellCheck: false, "data-gramm": false }),
diff --git a/src/index.tsx b/src/index.tsx
index 156a64b8b02d7c080acf15637f4e11677caa5894..47895ce9bc248ec6697535b916b3ee788d858f6a 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -554,6 +554,7 @@ export default class Editor extends React.Component<Props, State> {
         <pre
           className={preClassName}
           aria-hidden="true"
+          data-evaluate-disabled="true"
           style={{ ...styles.editor, ...styles.highlight, ...contentStyle }}
           {...(typeof highlighted === 'string'
             ? { dangerouslySetInnerHTML: { __html: highlighted + '<br />' } }
